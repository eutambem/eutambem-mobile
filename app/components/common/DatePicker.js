import React from 'react';
import ExternalDatePicker from 'react-native-datepicker';
import { StyleSheet } from 'react-native';

import appStyles, { RED, DARK_PURPLE } from '../../styles';

export const style = StyleSheet.create({
  btnTextCancel: {
    color: RED,
    fontSize: 18,
  },
  btnTextConfirm: {
    color: DARK_PURPLE,
    fontSize: 18,
  },
  date: {
    width: '100%',
    marginTop: 10,
    marginBottom: 10,
    height: 30,
    alignItems: 'flex-start',
    justifyContent: 'center',
  },
});

const DatePicker = props => (
  <ExternalDatePicker
    {...props}
    mode="date"
    format="DD/MMM/YYYY"
    confirmBtnText="Ok"
    cancelBtnText="Cancelar"
    showIcon={false}
    style={style.date}
    customStyles={{
      dateInput: appStyles.input,
      btnTextCancel: style.btnTextCancel,
      btnTextConfirm: style.btnTextConfirm,
    }}
  />
);

export default DatePicker;
