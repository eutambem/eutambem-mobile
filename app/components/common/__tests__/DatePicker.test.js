import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import DatePicker from '../DatePicker';

describe('DatePicker', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DatePicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
