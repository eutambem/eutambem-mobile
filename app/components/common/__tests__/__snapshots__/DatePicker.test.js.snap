// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<DatePicker
  TouchableComponent={[Function]}
  allowFontScaling={true}
  androidMode="default"
  cancelBtnText="Cancelar"
  confirmBtnText="Ok"
  customStyles={
    Object {
      "btnTextCancel": Object {
        "color": "#8C8C8C",
        "fontSize": 17,
      },
      "btnTextConfirm": Object {
        "color": "#961687",
        "fontSize": 17,
      },
      "dateInput": Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#91508c",
        "borderBottomWidth": 1,
        "borderWidth": 0,
        "height": 30,
        "margin": 0,
        "marginBottom": 10,
        "marginTop": 10,
      },
    }
  }
  date=""
  disabled={false}
  duration={300}
  format="DD/MMM/YYYY"
  height={259}
  hideText={false}
  iconSource={
    Object {
      "testUri": "../../react-native-datepicker/date_icon.png",
    }
  }
  modalOnResponderTerminationRequest={[Function]}
  mode="date"
  placeholder=""
  showIcon={false}
  style={
    Object {
      "alignItems": "flex-start",
      "height": 30,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
      "width": "100%",
    }
  }
/>
`;
